diag_manager/fms_diag_file_object.F90

```diff
@@ -0,0 +1,403 @@
+  procedure, public :: get_file_domain_type
+! TODO  procedure, public :: get_diag_yaml_file ! TODO
+  procedure, public :: get_file_metadata_from_model
+  procedure, public :: get_var_ids
+! The following fuctions come will use the yaml inquiry functions
+#ifdef use_yaml
+ procedure, public :: get_file_fname
+ procedure, public :: get_file_frequnit
+ procedure, public :: get_file_freq
+ procedure, public :: get_file_timeunit
+ procedure, public :: get_file_unlimdim
+!! TODO get functions for sub region stuff
+! procedure, public :: get_file_sub_region
+ procedure, public :: get_file_new_file_freq
+ procedure, public :: get_file_new_file_freq_units
+ procedure, public :: get_file_start_time
+ procedure, public :: get_file_duration
+ procedure, public :: get_file_duration_units
+ procedure, public :: get_file_varlist
+ procedure, public :: get_file_global_meta
+ procedure, public :: has_file_fname 
+ procedure, public :: has_file_frequnit 
+ procedure, public :: has_file_freq 
+ procedure, public :: has_file_timeunit 
+ procedure, public :: has_file_unlimdim 
+ procedure, public :: has_file_sub_region 
+ procedure, public :: has_file_new_file_freq 
+ procedure, public :: has_file_new_file_freq_units 
+ procedure, public :: has_file_start_time 
+ procedure, public :: has_file_duration 
+ procedure, public :: has_file_duration_units 
+ procedure, public :: has_file_varlist 
+ procedure, public :: has_file_global_meta 
+#endif
+
+end type fmsDiagFile_type
+
+type(fmsDiagFile_type), dimension (:), allocatable, target :: FMS_diag_files !< The array of diag files
+
+contains
+
+!< @brief Allocates the number of files and sets an ID based for each file
+!! @return true if there are files allocated in the YAML object
+logical function fms_diag_files_object_init ()
+#ifdef use_yaml
+ integer :: nFiles !< Number of files in the diag yaml
+ integer :: i !< Looping iterator
+ if (diag_yaml%has_diag_files()) then
+        nFiles = diag_yaml%size_diag_files()
+        allocate (FMS_diag_files(nFiles))
+        set_ids_loop: do i= 1,nFiles
+                FMS_diag_files(i)%diag_yaml_file => diag_yaml%diag_files(i)
+                FMS_diag_files(i)%id = i
+                allocate(FMS_diag_files(i)%var_ids(diag_yaml%diag_files(i)%size_file_varlist()))
+                allocate(FMS_diag_files(i)%var_index(diag_yaml%diag_files(i)%size_file_varlist()))
+                allocate(FMS_diag_files(i)%var_reg(diag_yaml%diag_files(i)%size_file_varlist()))
+                !! Initialize the integer arrays
+                FMS_diag_files(i)%var_ids = DIAG_NULL
+                FMS_diag_files(i)%var_reg = .FALSE.
+                FMS_diag_files(i)%var_index = DIAG_NULL
+        enddo set_ids_loop
+        fms_diag_files_object_init = .true.
+ else
+        fms_diag_files_object_init = .false.
+!        mpp_error("fms_diag_files_object_init: The diag_table.yaml file has not been correctly parsed.",&
+!                   FATAL)
+ endif
```

Comment:
Please fix the spacing here :/
https://github.com/NOAA-GFDL/FMS/blob/main/CODE_STYLE.md#general-1